docs/contributing/coding-style.md

```diff
@@ -1018,6 +1079,166 @@ Still, in some cases, you may need to change the generated code to meet your req
            // generated test code
    ```
 
+1. Context
+
+   If the target function accepts context as an input argument, the test code generated will include it in `args`.
+
+   e.g. 
+
+   ```go
+    type args struct {
+        ctx context.Context
+    }
+    ...
+   ```
+
+   But in test implementation, it is hard to manage the context lifecycle. We want to guarantee the context is closed after the test is executed, to avoid any missing termination of the process.
+
+   In Vald, we suggest customizing the test implementation from the generated test code to create the context and manage the lifecycle in every test.
+
+   We can remove the context from the `args` struct and create the context in the test execution code.
+
+   e.g.
+
+   ```go
+    // we can remove the ctx from args list
+    type args struct {
+        // ctx context.Context
+    }
+    ...
+
+    for _, tc := range tests {
+        test := tc
+        t.Run(test.name, func(tt *testing.T) {
+            // create context here
+            ctx, cancel := context.WithCancel(context.Background())
+            // handle the context lifecycle by canceling the context after the test is executed
+            defer cancel()
+
+            ...
+
+            // use the context created above
+            got, gotErr := someFunc(ctx)
+
+            ...
+   ```
+
+1. Struct initialization
+
+   By default, when testing the function of the struct, the target struct initialization is implemented by setting the date from the `fields` defined in the test case.
+   This initialization method has a few disadvatages:
+
+   1. When there are many fields in the struct, it is hard to set them all
+   2. The default value is the zero value of the type, not the struct default value from the struct initialization function
+
+   To resolve these problems, we can modify the test implementation to use the struct initialization function instead of setting the struct fields on the test cases.
+
+   For example, the current implementation may look like this:
```

Comment:
```suggestion
   By default, when testing the function of the struct, the target struct initialization is implemented by setting the date from the `fields` defined in the test case.
   This initialization method has a few disadvantages:

   1. When there are many fields in the struct, it is hard to set them all
   2. The default value is the zero value of the type, not the struct default value from the struct initialization function

   To resolve these problems, we can modify the test implementation to use the struct initialization function instead of setting the struct fields on the test cases.

   For example, the current implementation may look like this:
```